java/src/main/java/ai/rapids/cudf/Aggregation.java

```diff
@@ -396,7 +396,16 @@ private HostUDFAggregation(HostUDFWrapper wrapper) {
 
         @Override
         long createNativeInstance() {
-            return Aggregation.createHostUDFAgg(wrapper.udfNativeHandle);
+            long udf = 0;
+            try {
+                udf = wrapper.createUDFInstance();
+                return Aggregation.createHostUDFAgg(udf);
```

Comment:
UDF instance is hold by Aggregation instance via a `unique_ptr`,  the Aggregation instance is responsible for the life cycle of UDF instance.
Here pass in a UDF instance and `createHostUDFAgg` clone it and construct a Aggregation instance, at the finally block, we close the original UDF instance.